src/components/AnnotationDocument.tsx

```diff
@@ -0,0 +1,39 @@
+import { domainAndTitle } from '../helpers/annotation-metadata';
+import type { Annotation } from '../helpers/annotation-metadata';
+import AnnotationDocumentInfo from './AnnotationDocumentInfo';
+
+export type AnnotationDocumentProps = {
+  annotation: Annotation;
+  pageNumber?: string;
+};
+
+/**
+ * Renders annotation document information, like document title, document URL
+ * and a page number if provided.
+ */
+export default function AnnotationDocument({
+  annotation,
+  pageNumber,
+}: AnnotationDocumentProps) {
+  const documentInfo = domainAndTitle(annotation);
+  const annotationURL = annotation.links?.html || '';
+  const documentLink =
+    annotationURL && documentInfo.titleLink ? documentInfo.titleLink : '';
+
+  return (
+    <span className="flex">
+      {documentInfo.titleText && (
+        <AnnotationDocumentInfo
+          domain={documentInfo.domain}
+          link={documentLink}
+          title={documentInfo.titleText}
+        />
```

Comment:
`AnnotationDocumentInfo` is not used outside of this component. Perhaps these changes could just be folded into `AnnotationDocumentInfo`?

Reply:
Yep, makes sense. I was trying to keep an internal annotation-decoupled component, and then an annotation-aware wrapper, but it's true it's probably overkill in this case.

On top of that, I have just realized we need to be able to render the page number i cases where the rest of the annotation info does not need to render.

I was working around that by duplicating a bit of logic, but I think I'm just going to remove the notion of the page number entirely for now, and just bring the document-related helpers and use them from the existing `AnnotationDocumentInfo`.